File: Packages/AuralisCore/Sources/Domain/QueueEntry.swift
import Foundation

/// 播放队列项（R05）：歌曲身份（Track）与队列项身份（UUID）分离。
///
/// 同一首歌可以在队列中出现多次，每次都是独立的队列项（对应 OpenSubsonic
/// `indexBasedQueue` 语义：队列按 index 定位，当前曲目由 index 表示）。
/// SwiftUI 列表 / ForEach 必须用 `id`（UUID）作为身份，不能用 `track.id`——
/// 否则重复歌曲会在渲染期因重复 ID 崩溃。
public struct QueueEntry: Codable, Hashable, Sendable, Identifiable {
    public let id: UUID
    public var track: Track

    public init(id: UUID = UUID(), track: Track) {
        self.id = id
        self.track = track
    }
}

File: android/core/domain/src/main/java/com/auralis/core/domain/QueueEntry.kt
package com.auralis.core.domain

import java.util.UUID

/**
 * 播放队列项（P0 语义）。
 *
 * 歌曲身份（[Track]）与队列项身份（[id]）**必须分离**：
 * 同一首歌可以在队列中出现多次，每一次 occurrence 都是独立队列项
 * （对应 OpenSubsonic `indexBasedQueue`：队列按 index 定位）。
 *
 * 因此：
 * - MediaItem.mediaId = [id].value（**不是** track.id）
 * - 点击第 N 个 occurrence 必须精确播放第 N 个
 * - 上一首/下一首基于 entry id / 逻辑下标，绝不能「找第一个 TrackID」
 */
data class QueueEntry(
    val id: QueueEntryId = QueueEntryId(UUID.randomUUID().toString()),
    val track: Track,
) {
    companion object {
        fun of(track: Track) = QueueEntry(track = track)
    }
}
